src/server/hset_family.cc

```diff
@@ -1020,24 +1050,90 @@ void HSetFamily::HRandField(CmdArgList args, ConnectionContext* cntx) {
     StringVec str_vec;
 
     if (pv.Encoding() == kEncodingStrMap2) {
-      // TODO: to create real random logic.
       StringMap* string_map = (StringMap*)pv.RObjPtr();
+      if (args.size() == 1) {
+        sds key;
+        string_map->RandomPair(&key, nullptr);
+        // sds key = string_map->begin()->first;
+        str_vec.emplace_back(key, sdslen(key));
+      } else {
+        size_t actual_count =
+            (count >= 0) ? std::min(size_t(count), string_map->Size()) : abs(count);
+        sds *keys = nullptr, *vals = nullptr;
+        keys = new sds[actual_count];
+        if (!with_values.empty())
+          vals = new sds[actual_count];
+        if (count >= 0) {
+          if (!with_values.empty()) {
+            string_map->RandomPairsUnique(actual_count, keys, vals);
+          } else {
+            string_map->RandomPairsUnique(actual_count, keys, nullptr);
```

Comment:
1. It's already null, why the if
2. Lets don't use raw arrays and don't pass nullptr. The performance isn't gonna measurably drop if we return the values as well

Reply:
> 1. It's already null, why the if

I don't understand this comment, you mean `vals` is already null? vals will be null if `WITHVALUES` is not specified. Otherwise, vals will be allocated through:

`
        if (!with_values.empty())
          vals = new sds[actual_count];
`

> 2\. Lets don't use raw arrays and don't pass nullptr. The performance isn't gonna measurably drop if we return the values as well

It was not for performance, it was a way to allow us fill vals only when `WITHVALUES` is specified. But I will change to use vector and pass in another boolean value with_values to achieve the same purpose.

Reply:
You can always return an empty vector if the `WITHVALUES` is not specified

Reply:
yes, but now this `with_values` has been changed to a boolean value, it is much more clear and readable. Thanks!